src/main.rs

```diff
@@ -246,17 +225,59 @@ fn migrate(data_string: String) -> Result<String, anyhow::Error> {
     Ok(output_path_str.to_string())
 }
 
-#[launch]
-fn rocket() -> Rocket<Build> {
+async fn rm_file_after_expiration(database: &Arc<Mutex<Connection>>) -> Result<(), anyhow::Error> {
+    let database = database.lock().await;
+    let mut stmt = database.prepare(format!("SELECT * FROM {}", TABLE_NAME).as_str())?;
+    let rows = stmt.query([])?;
+    let rows = rows.mapped(|row| {
+        Ok(Entry {
+            uuid: row.get(0)?,
+            file_path: row.get(1)?,
+            creation_time: row.get(2)?,
+        })
+    });
+    for row in rows {
+        let row = row?;
+        let time_passed_since_creation = chrono::Local::now().time()
+            - chrono::DateTime::parse_from_str(row.creation_time.as_str(), TIME_FORMAT)?.time();
+        let t_delta = match time_passed_since_creation
+            .checked_sub(&chrono::TimeDelta::new(FILE_EXPIRATION_IN_SEC, 0).unwrap())
+        {
+            Some(t_delta) => t_delta,
+            None => chrono::TimeDelta::zero(),
+        };
+        if t_delta > chrono::TimeDelta::zero() {
+            std::fs::remove_file(row.file_path)?;
+
+            let mut stmt = database
+                .prepare(format!("DELETE FROM {} WHERE uuid = (?1)", TABLE_NAME).as_str())?;
+            stmt.execute([row.uuid])?;
+        }
+    }
+    Ok(())
+}
+
+async fn async_db_cleanup(db_clone: Arc<Mutex<Connection>>) {
+    loop {
+        match rm_file_after_expiration(&db_clone).await {
+            Ok(ok) => ok,
+            Err(e) => println!("{}", e),
+        };
+        rocket::tokio::time::sleep(std::time::Duration::from_secs(3*60)).await;
```

Comment:
3 minutes is a bit of an odd sleep duration, if the expiration timer is 5 minutes and could result in a pretty long time before a file is cleaned up although it's already 5 minutes old. It isn't a heavy operation so I would lean towards a pretty short timer, something like 15-30 seconds.